### PR TITLE
Make Login, Logout and ChangePassword paths configurable

### LINKED ISSUE #6019: Custom Login, Logout, ChangePassword path
Since we have AdminOptions now, can we make Login, Logout and ChangePassword routes customizable also?

I have a site that has its own user portal.  The links are in conflict with the default users login in orchard.  Before endpoint routing, I could just hijack the links by replacing them in configure.  But I couldn't easily do that with endpoint routing. (Upgrading from a very old OrchardCore to latest)

I could do a PR if this is going to be merged.

I plan to just add the 3 paths into AdminOptions.  Please let me know.

I need this asap.  Thanks!

### PR DESCRIPTION
**Pull request by awyl**
Fixes #6019 

There are other hardcoded paths in Users.  Do we want to make them configurable as well?

### CODE REVIEW

**Comment by agriffard** [comment]
Can you please add an example to the doc?
In a src\docs\reference\modules\Users\README.md file similar to src\docs\reference\modules\Admin\README.md

**Comment by awyl** [comment]
> Can you please add an example to the doc?
> In a src\docs\reference\modules\Users\README.md file similar to src\docs\reference\modules\Admin\README.md

Done.  Please check my english and grammar.

File: src/docs/reference/modules/Users/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+## Custom Paths
+
+If you want to specify custom paths to login page, logoff, etc...
+
+Below listed the configurable paths with their default values.  Specify the one you want to configure in appsettings.json:
+
+``` json
+  "OrchardCore": {
+    "OrchardCore_Admin": {
```
Comment (agriffard) [inline src/docs/reference/modules/Users/README.md:13]: OrchardCore_User

File: src/docs/reference/modules/Users/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Users (OrchardCore.Users)
+
+The Users module enables authentication UI and user management.
+
+## Custom Paths
+
+If you want to specify custom paths to login page, logoff, etc...
```
Comment (agriffard) [inline src/docs/reference/modules/Users/README.md:7]: If you want to specify custom paths to access the authentication related urls, you can change ithem by using this option in the appsettings.json:

File: src/docs/reference/modules/Users/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+Below listed the configurable paths with their default values.  Specify the one you want to configure in appsettings.json:
```
Comment (agriffard) [inline src/docs/reference/modules/Users/README.md:9]: Remove

File: src/docs/reference/modules/Users/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+``` json
+  "OrchardCore": {
+    "OrchardCore_Admin": {
+      "LoginPath": "Login",
+      "LogoffPath": "/Users/LogOff",
```
Comment (agriffard) [inline src/docs/reference/modules/Users/README.md:15]: /Users/LogOff => LogOff

File: src/OrchardCore/OrchardCore.Users.Abstractions/UserOptions.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        private string _loginPath = "Login";
+        private string _logoffPath = "/Users/LogOff";
+        private string _changePasswordUrl = "ChangePassword";
+
+        public string LoginPath
+        {
+            get => _loginPath;
+            set => _loginPath = value.Trim();
+        }
```
Comment (deanmarcussen) [inline src/OrchardCore/OrchardCore.Users.Abstractions/UserOptions.cs:13]: Probably here it is best to trim, as we did with the `AdminOptions` 
` _loginPath = value.Trim(' ', '/');

File: src/OrchardCore.Modules/OrchardCore.Users/Startup.cs [hunk 6]
```diff
@@ -111,6 +113,9 @@ public override void Configure(IApplicationBuilder builder, IEndpointRouteBuilde
 
         public override void ConfigureServices(IServiceCollection services)
         {
+            // services.Configure<UserOptions>(_configuration.GetSection("OrchardCore_Users"));
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Users/Startup.cs:116]: You can remove this commented line

File: src/OrchardCore/OrchardCore.Users.Abstractions/UserOptions.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
+namespace OrchardCore.Users
+{
+    public class UserOptions
+    {
+        private string _loginPath = "Login";
+        private string _logoffPath = "/Users/LogOff";
+        private string _changePasswordUrl = "ChangePassword";
```
Comment (deanmarcussen) [inline src/OrchardCore/OrchardCore.Users.Abstractions/UserOptions.cs:7]: Not sure if this needs leading slash (I know it currently has one)

Can you check if it is good without, and if so remove it, and the trim can then be the same across them all?

Comment (awyl) [reply src/OrchardCore/OrchardCore.Users.Abstractions/UserOptions.cs:7]: It works without the leading /.

File: src/docs/reference/modules/Users/README.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# Users (OrchardCore.Users)
+
+The Users module enables authentication UI and user management.
+
+## Custom Paths
+
+If you want to specify custom paths to access the authentication related urls, you can change ithem by using this option in the appsettings.json:
```
Comment (deanmarcussen) [inline src/docs/reference/modules/Users/README.md:7]: ithem -> them

File: src/docs/reference/modules/Users/README.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+``` json
+  "OrchardCore": {
+    "OrchardCore_Users": {
+      "LoginPath": "Login",
+      "LogoffPath": "LogOff",
```
Comment (deanmarcussen) [inline src/docs/reference/modules/Users/README.md:13]: the default is `Users/LogOff' if I am correct?

Comment (awyl) [reply src/docs/reference/modules/Users/README.md:13]: Lol, half the change from you and agriffard.

File: mkdocs.yml [hunk 10]
```diff
@@ -130,6 +130,7 @@ nav:
                 - Twitter: docs/reference/modules/Twitter/README.md
                 - GitHub: docs/reference/modules/GitHub/README.md
                 - Google: docs/reference/modules/Google/README.md
+            - Users: docs/reference/modules/Users/README.md
```
[the remark this answers is not in the document]
Comment (awyl) [reply mkdocs.yml:133]: It should nest under Authentication?

**Comment by awyl** [comment]
@jtkech Thanks for the detail explanation.  I have updated the code.

File: mkdocs.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (jtkech) [reply mkdocs.yml:133]: Oups sorry, you're right

**Pull request merged**